src/assets/i18n/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { I18n } from './types';
+import { sprintf } from 'sprintf-js';
+import en from './en';
+
+type Language = keyof typeof dictionary;
+
+const dictionary = { en };
+
+let language: Language = 'en';
+
+export function setLanguage(lang: Language) {
+  language = lang;
+}
+
+export function getLanguage() {
+  return language;
+}
+
+/**
+ * Returns a formatted i18n string using sprintf-js.
+ *
+ * @param key The key of an i18n string.
+ * @param args The arguments to format the string with.
+ */
+export function t(key: keyof I18n, ...args: any[]) {
```

Comment:
Is this "t" for "translate"? If so, perhaps just use the full word for ease of reading?

Reply:
Yes, and I brought the idea from [react-i18next](https://github.com/i18next/react-i18next). Since this function would be used very often in both .html and .ts, I think it would rather improve the overall readability if its name is as short as possible.